tests/test_ote_training.py

```diff
@@ -0,0 +1,690 @@
+
+    if isinstance(some_val, str):
+        if not osp.isabs(some_val):
+            return osp.join(root_path, some_val)
+        return some_val
+
+    some_dict = some_val
+    for k in sorted(some_dict.keys()):
+        v = some_dict[k]
+        if isinstance(v, str) and not osp.isabs(v):
+            some_dict[k] = osp.join(root_path, v)
+    return some_dict
+
+def _get_dataset_params_from_dataset_definitions(dataset_definitions, dataset_name):
+    cur_dataset_definition = dataset_definitions[dataset_name]
+    training_parameters_fields = {k: v for k, v in cur_dataset_definition.items()
+                                  if k in DATASET_PARAMETERS_FIELDS()}
+    _make_path_be_abs(training_parameters_fields, dataset_definitions[ROOT_PATH_KEY])
+
+    assert set(DATASET_PARAMETERS_FIELDS()) == set(training_parameters_fields.keys()), \
+            f'ERROR: dataset definitions for name={dataset_name} does not contain all required fields'
+    assert all(training_parameters_fields.values()), \
+            f'ERROR: dataset definitions for name={dataset_name} contains empty values for some required fields'
+
+    params = DatasetParameters(**training_parameters_fields)
+    return params
+
+def to_dict(val):
+    """
+    General function to convert any class/structure to a serializable dict
+    if the class/structure is sufficiently simple; but nested simple structs are possible.
+    """
+    import numpy as np
+    def _to_dict(val):
+        PRIMITIVE_TYPES = (bool, int, float, str, complex) #
+        if type(val) in PRIMITIVE_TYPES:
+            return val
+        if isinstance(val, np.number):
+            return val
+        if type(val) == list:
+            return [_to_dict(v) for v in val]
+        if type(val) == dict:
+            keys = list(val.keys())
+            return {k: _to_dict(v) for k, v in val.items()}
+
+        keys = [k for k in dir(val) if not k.startswith('_') and not callable(getattr(val, k))]
+        res = {}
+        for k in keys:
+            res[k] = _to_dict(getattr(val, k))
+        return res
+    res = _to_dict(val)
+    return res
+
+def performance_to_score_name_value(perf: Performance):
+    """
+    The method is intended to get main score info from Performance class
+    """
+    if perf is None:
+        return None, None
+    assert isinstance(perf, Performance)
+    score = perf.score
+    assert isinstance(score, ScoreMetric)
+    name = score.name
+    value = score.value
+    return name, value
+
+def select_configurable_parameters(json_configurable_parameters):
+    selected = {}
+    getv = lambda c, n: c[n]['value']
+    for section, container in json_configurable_parameters.items():
+        for param in container.keys():
+            try:
+                selected[f'{section}/{param}'] = getv(container, param)
+            except TypeError:
+                pass
+            except KeyError:
+                pass
+    return selected
+
+def convert_hyperparams_to_dict(hyperparams):
```

Comment:
does ote_sdk.configuration.helper.convert do the same job, can it be used?

Reply:
I will check it

Reply:
Ilya, I checked it.

Unfortunately, `ote_sdk.configuration.helper.convert` creates dicts with a lot of metadata info, the values of the parameters are written into these dicts as one more key-value pair  with key `value`.
But I want to get from hyperparams a short info like the following:
```python
{
 'learning_parameters': {'batch_size': 2,
                         'learning_rate': 0.008,
                         'learning_rate_warmup_iters': 200,
                         'num_checkpoints': 2,
                         'num_iters': 1,
                         'num_workers': 0},
 'postprocessing': { # some data...
  },
# ... etc ...
}
```
To extract such info from the result of `ote_sdk.configuration.helper.convert`  we need to make additional function that uses heuristics to extract only fields with key `value`
I wrote such function and it requires more lines of code than the current implementation. 
At the same time it makes the code less clear -- instead of recursive walk on attribute `groups`, we have recursive walk on dict of dicts with some strange heuristic.

So, I prefer to keep the current decision.

Reply:
BTW, the function `parameter_group_to_dict` from `ote_sdk/configuration/helper/convert.py` makes absolutely the same recursive walk as my function `_convert`, but adds metainfo and formats result in a way that is not convenient for me.